src/transport_performance/utils/defence.py

```diff
@@ -232,36 +233,108 @@ def _type_defence(some_object, param_nm, types) -> None:
     return None
 
 
-def _check_list(ls, param_nm, check_elements=True, exp_type=str):
-    """Check a list and its elements for type.
+def _check_iter_length(iterable: Iterable, param_nm: str, length: int):
+    """Check the length of an iterable.
 
     Parameters
     ----------
-    ls : list
-        List to check.
+    iterable : Iterable
+        Iterable to check.
     param_nm : str
         Name of the parameter being checked.
-    check_elements : (bool, optional)
-        Whether to check the list element types. Defaults to True.
-    exp_type : (_type_, optional):
-        The expected type of the elements. Defaults to str.
+    length: int
+        Expected length of the iterable to check.
 
     Raises
     ------
-        TypeError: `ls` is not a list.
-        TypeError: Elements of `ls` are not of the expected type.
+        ValueError: length of iterable does not match `length`.
 
     Returns
     -------
     None
 
     """
-    if not isinstance(ls, list):
-        raise TypeError(
-            f"`{param_nm}` should be a list. Instead found {type(ls)}"
+    # check if iterable
+    _type_defence(iterable, param_nm, Iterable)
+    # check if length is int
+    _type_defence(length, "length", int)
+
+    if len(iterable) != length:
+        raise ValueError(
+            f"`{param_nm}` is of length {len(iterable)}. "
+            f"Expected length {length}."
         )
+
+    return None
+
+
+def _check_iterable(
+    iterable: Iterable,
+    param_nm: str,
+    iterable_type: type,
+    check_elements: bool = True,
+    exp_type: Union[tuple, type] = str,
+    check_length: bool = False,
+    length: int = 0,
+):
+    """Check an iterable and its elements for type.
+
+    Parameters
+    ----------
+    iterable : Iterable
+        Iterable to check.
+    param_nm : str
+        Name of the parameter being checked.
+    iterable_type : type
+        Expected iterable type.
+    check_elements : bool, optional
+        Whether to check the list element types. Defaults to True.
+    exp_type : Union[tuple, type], optional:
+        The expected type of the elements. Defaults to str. If using a tuple,
+        it should be a tuple of types.
+    check_length: bool, optional
+        Wether to check the length of the iterable. Defaults to False.
+    length: int, optional
+        Expected length of the iterable.
+
+    Raises
+    ------
+        TypeError: `iterable` is not iterable.
+        TypeError: `exp_type` contains elements that are not type.
+        TypeError: Elements of `iterable` are not of the expected type(s).
```

Comment:
This section should follow the Numpy style guide ([see raises section here](https://numpydoc.readthedocs.io/en/latest/format.html#raises)).

An example of this would be:  
```python
"""
Raises
------
TypeError:
    'iterable' is not iterable.
TypeError:
    etc
TypeError:
    etc

"""
```

Reply:
Changed as suggested following numpydoc style.